File: ecommerce-go-fiber/apps/products/internal/config/env.go
package config

import (
	"os"
)

type Env struct {
	ProductPort        string
	ProductMongoURI    string
	ProductServiceName string
	GRPCServerPort     string
}

func required(name string, fallback string) string {
	value := os.Getenv(name)
	if value == "" {
		return fallback
	}
	return value
}

func getString(name, fallback string) string {
	value := os.Getenv(name)
	if value == "" {
		return fallback
	}

	return value
}

func GetEnv() Env {
	return Env{
		ProductPort:        getString("PRODUCT_PORT", "4030"),
		ProductMongoURI:    required("PRODUCT_MONGO_URI", "mongodb://admin:password@localhost:27018/products?authSource=admin"),
		ProductServiceName: getString("SERVICE_NAME", "products"),
		GRPCServerPort:     getString("GRPC_SERVER_PORT", "50052"),
	}
}

File: ecommerce-ts-express/services/products/src/config/env.ts
export type Env = {
  productPort: string
  productMongoUri: string
  productServiceName: string
  grpcHost: string
}

function required(name: string, fallback: string): string {
  const value = process.env[name]
  if (!value) {
    return fallback
  }

  return value
}

function getString(name: string, fallback: string): string {
  const value = process.env[name]
  if (!value) {
    return fallback
  }

  return value
}

export function getEnv(): Env {
  return {
    productPort: getString('PORT', '3030'),
    productMongoUri: required(
      'MONGO_URI',
      'mongodb://admin:password@localhost:27018/products?authSource=admin'
    ),
    productServiceName: getString('NAME', 'products'),
    grpcHost: getString('GRPC_HOST', ':50051')
  }
}
